The following is a fix for https://youtrack.jetbrains.com/issue/IDEA-383732
while we wait for the upstream fix to become available.

diff --git a/java/java-psi-impl/src/com/intellij/psi/impl/file/PsiPackageImpl.java b/java/java-psi-impl/src/com/intellij/psi/impl/file/PsiPackageImpl.java
index aa8d14da59d2..560cb51d9658 100644
--- a/java/java-psi-impl/src/com/intellij/psi/impl/file/PsiPackageImpl.java
+++ b/java/java-psi-impl/src/com/intellij/psi/impl/file/PsiPackageImpl.java
@@ -441,7 +441,13 @@ public class PsiPackageImpl extends PsiPackageBase implements PsiPackage, Querya
       }
 
       for (PsiClass aClass : getFacade().findClasses(getQualifiedName() + ".package-info", allScope())) {
-        ContainerUtil.addIfNotNull(modifiers, aClass.getModifierList());
+        PsiFile file = aClass.getContainingFile();
+        if (file instanceof PsiJavaFile) {
+          PsiPackageStatement stmt = ((PsiJavaFile)file).getPackageStatement();
+          if (stmt != null) {
+            ContainerUtil.addIfNotNull(modifiers, stmt.getAnnotationList());
+          }
+        }
       }
 
       PsiCompositeModifierList result = modifiers.isEmpty() ? null : new PsiCompositeModifierList(getManager(), modifiers);
